numpy/core/src/multiarray/ctors.c

```diff
@@ -1282,7 +1282,7 @@ _array_from_array_like(PyObject *op,
             PyErr_Clear();
         }
         else {
-            tmp = _array_from_buffer_3118(memoryview);
+            tmp = _array_from_buffer_3118(memoryview); // Assume: never creates a copy
```

Comment:
This is technically not necessarily true.  Python has some very ugly code to try and guess if it is true for its argument parsing with `"s"`.

I am not sure what to do about it though, maybe we can get away with assuming that it is true.  Ideally it would always be true (or _almost_ always, since it would make sense to add suboffsets to NumPy in the form of a copy!)

This may be something to discuss briefly.